src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs

```diff
@@ -107,6 +109,25 @@ bool isStatic
             return null;
         }
 
+        if (BuildGenericTypeParameters(methodSymbol, parameters, out var typeParameters))
+        {
+            return new UserDefinedNewInstanceGenericTypeMapping(
+                methodSymbol,
+                typeParameters.Value,
+                parameters,
+                ctx.MapperConfiguration.UseReferenceHandling,
+                ctx.Types.PreserveReferenceHandler,
+                GetTypeSwitchNullArm(methodSymbol, parameters, typeParameters),
+                ctx.Compilation.ObjectType
+            );
+        }
+
+        if (methodSymbol.IsGenericMethod)
+        {
+            ctx.ReportDiagnostic(DiagnosticDescriptors.UnsupportedMappingMethodSignature, methodSymbol, methodSymbol.Name);
```

Comment:
Is there a test for this? Codecov seems to complain

Reply:
Added a test.